catroid/src/main/java/org/catrobat/catroid/camera/PreviewView.kt

```diff
@@ -41,31 +40,13 @@ class PreviewView(context: Context) : FrameLayout(context) {
     fun createSurfaceProvider() = SurfaceProvider { request ->
         with(request.resolution) {
             surfaceView.holder.setFixedSize(width, height)
-            scaleView(width, height)
+            surfaceView.scaleX = Resolution(width, height).aspectRatio /
+                Resolution(this.width, this.height).aspectRatio
         }
         request.provideSurface(
             surfaceView.holder.surface,
             Executors.newSingleThreadExecutor(),
             Consumer { }
         )
     }
-
-    private fun scaleView(imageWidth: Int, imageHeight: Int) {
-        val imageAspectRatio = imageHeight.toFloat() / imageWidth
-        val screenAspectRatio = this.width.toFloat() / this.height
-
-        if (screenAspectRatio < 1) { // portrait mode
-            val scalingFactor = imageAspectRatio / screenAspectRatio
-            if (Build.VERSION.SDK_INT == Build.VERSION_CODES.M) {
```

Comment:
@mseiser do you know if the special behavior for SDK 23(Android M) is needed?

Reply:
Our MinSDK is still SDK 23, please have a look at the AndroidManifest.xml, regarding SDK requirements